### PR TITLE
dev/eahw-2109/publish-jparest-library: Implement drone pipeline inclu…

### PR DESCRIPTION
**Pull request by JWeston-HO**
…ding snapshots and tags

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -33,9 +33,55 @@ steps:
   - mvn sonar:sonar -Dsonar.host.url=$${SONAR_HOST} -Dsonar.login=$${SONAR_TOKEN} -DprojectKeySuffix=$${projectSuffix}
   when:
     event:
-      exclude:
+      exlcude:
```
Comment (laurieboyes) [inline .drone.yml:36]: Typo?
```suggestion
      exclude:
```

File: .drone.yml [hunk 2]
```diff
@@ -33,9 +33,55 @@ steps:
...
+  commands:
+  - mvn -s ./jparest_settings.xml -f pom.xml install -B org.apache.maven.plugins:maven-deploy-plugin:2.8.2:deploy -DskipTests
+  when:
+    branch: main
+#    event:
+#      exlcude:
+#      - pull_request
+
+- name: Git Tag
+  image: jlrigau/maven-git
```
Comment (laurieboyes) [inline .drone.yml:55]: This project doesn’t look to have been recently maintained 🤔 How confident are we that we should use it? Is there a fresher alternative?

File: .drone.yml [hunk 3]
```diff
@@ -33,9 +33,55 @@ steps:
...
 
+- name: snapshot
+  image: maven:3.8-openjdk-17
+  depends_on:
+    - sonar
+  environment:
+    ARTIFACTORY_TOKEN:
+      from_secret: artifactory_token
+  commands:
+    - "BRANCH=$(echo $DRONE_BRANCH | tr '[:lower:]' '[:upper:]') && [[ $BRANCH =~ (EAHW-[0-9]*) ]] && snapshotSuffix=$${match/#E/-E}$${BASH_REMATCH/#E/-E}"
```
Comment (laurieboyes) [inline .drone.yml:78]: Is there somewhere we can see an example of the snapshot name this command generates?

File: pom.xml [hunk 4]
```diff
@@ -10,6 +10,8 @@
     <packaging>pom</packaging>
 
     <properties>
+        <project.version>0.0.0${snapshotSuffix}</project.version>
```
Comment (laurieboyes) [inline pom.xml:13]: Do we have to remember to update this version number when we make a change? I assume yes. Might be worth an entry in the readme 🤔

**Review by laurieboyes — COMMENTED** [review]
Looking pretty good! A few comments

**Review by laurieboyes — APPROVED** [review]
Beautifully honed

File: .drone.yml [hunk 5]
```diff
@@ -36,6 +36,56 @@ steps:
...
 
+- name: publish
+  image: maven:3.8-openjdk-17
+  depends_on:
+    - sonar
+  environment:
+    ARTIFACTORY_TOKEN:
+        from_secret: artifactory_token
+  commands:
+  - mvn -s ./jparest_settings.xml -f pom.xml install -B org.apache.maven.plugins:maven-deploy-plugin:2.8.2:deploy -DskipTests
```
Comment (daleelR) [inline .drone.yml:47]: Is it ok to skip the test run here. Wouldn't we want to ensure they all pass first?

Comment (JWeston-HO) [reply .drone.yml:47]: The unit tests run under the test step. Didn't want to add to build time by running unit tests twice.

Comment (daleelR) [reply .drone.yml:47]: Ah right, yep makes sense.

**Review by daleelR — APPROVED** [review]

**Pull request merged**